### PR TITLE
Remove DB check for receiving p2p slashings

### PR DESCRIPTION
**Pull request by 0xKiwi**
**What type of PR is this?**
Bug fix

**What does this PR do? Why is it needed?**
This PR fixes a bug regarding beacon node receiving broadcasted P2P slashings. Currenty we use `db.State` on the slashing's first element block root, but this is not needed and prevents any slashing older than 3 epochs from being accepted. This PR removes those checks.

### CODE REVIEW

File: beacon-chain/sync/subscriber_handlers.go [hunk 1]
```diff
@@ -31,46 +29,45 @@ func (r *Service) voluntaryExitSubscriber(ctx context.Context, msg proto.Message
...
 	if !ok {
 		return fmt.Errorf("wrong type, expected: *ethpb.ProposerSlashing got: %T", msg)
 	}
 	// Do some nil checks to prevent easy DoS'ing of this handler.
-	if ps.Header_1 != nil && ps.Header_1.Header != nil {
-		root, err := stateutil.BlockHeaderRoot(ps.Header_1.Header)
-		s, err := r.db.State(ctx, root)
+	header1IsNil := pSlashing == nil || pSlashing.Header_1 == nil || pSlashing.Header_1.Header == nil
+	header2IsNil := pSlashing == nil || pSlashing.Header_2 == nil || pSlashing.Header_2.Header == nil
+	if !header1IsNil && !header2IsNil {
```
Comment (shayzluf) [inline beacon-chain/sync/subscriber_handlers.go:60]:
```suggestion
        pSlashingNil := pSlashing == nil
	header1IsNil := pSlashing.Header_1 == nil || pSlashing.Header_1.Header == nil
	header2IsNil := pSlashing.Header_2 == nil || pSlashing.Header_2.Header == nil
	if !pSlashingNil && !header1IsNil && !header2IsNil {
```

Comment (0xKiwi) [reply beacon-chain/sync/subscriber_handlers.go:60]: This would be nice, but I think it'll cause nil pointer if pSlashing is nil, gonna leave as is.

Comment (nisdas) [reply beacon-chain/sync/subscriber_handlers.go:60]: Yeah that will cause a panic in the event `pSlashing` is empty

File: beacon-chain/sync/subscriber_handlers.go [hunk 2]
```diff
@@ -31,46 +29,45 @@ func (r *Service) voluntaryExitSubscriber(ctx context.Context, msg proto.Message
...
-			return fmt.Errorf("no state found for block root %#x", root)
-		}
-		if err := r.slashingPool.InsertProposerSlashing(ctx, s, ps); err != nil {
+		if err := r.slashingPool.InsertProposerSlashing(ctx, headState, pSlashing); err != nil {
 			return errors.Wrap(err, "could not insert proposer slashing into pool")
 		}
-		r.setProposerSlashingIndexSeen(ps.Header_1.Header.ProposerIndex)
+		r.setProposerSlashingIndexSeen(pSlashing.Header_1.Header.ProposerIndex)
+	} else {
+		return errors.New(fmt.Sprintf("h1: %t, h2: %t", header1IsNil, header2IsNil))
```
Comment (nisdas) [inline beacon-chain/sync/subscriber_handlers.go:70]: why return this ?

Comment (0xKiwi) [reply beacon-chain/sync/subscriber_handlers.go:70]: Whooooops thank you

File: beacon-chain/sync/subscriber_handlers.go (hunk 1 above)
Comment (shayzluf) [reply beacon-chain/sync/subscriber_handlers.go:60]: yeh, too many lines for this to work. keeping as it is is great

**Pull request merged**